### ISSUE TITLE
Front-end users can't upload media

### ISSUE THREAD

**Issue by hawkwynd** [wpmudev/wiki#1]
If a front-end user (Non WP-admin) is logged in, the wiki editor does not display the "Add Media" button.

**Comment by kalico1** [comment]
In my case, front-end user has no editing privileges at all. The entire right-side menu is not there, no matter what I've tried for permissions so far, for both the user and for wiki editing. I'm using Twenty-Fifteen, so it shouldn't be a theme conflict. I'll keep looking for a solution! May end up forking this and working on it myself, as there is no WPMUDEV support for these anymore. <sad face>

**Comment by kalico1** [comment]
I solved my problem by giving the Contributor and Author roles all the "wiki" related permissions, which were not being set automatically as I would have expected. (I used the User Role Editor plugin to accomplish it.) I might fiddle with this to fine tune what each role can do, but this is the general solution for the problem I cited.

I still have a problem sorting out the options on a per-wiki-page basis. There are 4 options -- and the only one that seems to allow my Contributors and Authors to edit on the frontend is "Anyone" which isn't really ideal for me. I do have other ways of protecting the content from editing, so mayve I'll just go that route. But I'd rather have it working right.....

Based on what I learned through this, I have a suggestion for you @hawkwynd : **try making your front-end user an Author.** By default, Contributors and below do not have the capability to upload media. I am still trying to figure out exactly how to give Contributors this cap, but all I've found is a plugin (seems like overkill) or custom code (why? isn't there a standard WP cap for this?). I'll post back when I find the answer.

**Comment by hawkwynd** [comment]
@kalico1 - Activating the "WP Admin" account when creating the user solved the issue, and I set their WP account to 'Author' which gave them the privelages they needed to use the "Add Media" controls. Thank you! While it took me hours and hours of digging to find this "little-known" fact in the documentation/google searching - I'm relieved it's working for the client now.

**Comment by kalico1** [comment]
Hi @hawkwynd - Can you clarify something? You mentioned here that "activating the WP Admin account when creating the user solved the issue" but you also had to set them as Author. Do you mean that initially, you set their role to Admin, then later changed to Author? Or are you referring to two separate things that both worked out the same? Sorry if I'm misreading this, I'm just curious to because I'm trying to document the settings that are required, and I don't want to miss something if your results were different from mine. Thanks!

**Comment by hawkwynd** [comment]
@kalico1 To clarify: 
We're using a plugin named "Front End Users" http://www.etoilewebdesign.com/ to allow users to login to the website to access/edit/update WIKI posts.  In the settings for FEU on the Premium tab, I enabled the setting "Create WordPress User".  When a new user is created in FEU, it creates a WP Admin account as well. After the account is created, I changed the ROLE of that user in WP Users to from the default to "Author". Front-End Users then log in using the assigned email address for their account, and within the WIKI they are able use the "Add Media" options. By default, WP does not allow uploads without a WP account login session. Hope this clarifies things better for you.
